### PR TITLE
Repository import event job

### PR DESCRIPTION
**Pull request by BenEmdon**
## What
This PR proposes a new approach for reporting a `AssignmentRepo`'s starter code import as opposed to our current offering (`PorterStatusJob`).
Instead of polling and for a source imports status, instead we will kick off a job on the `repository_import` event.

The pros and cons of this approach were discussed in https://github.com/education/classroom/issues/1498

## Why is this now a priority?
This should reduce the load of requests that have caused throttling incidents (#1543 #1542 #1537 #1536)

## How will this affect UX 
We will no longer get intermediate steps form the import process and will instead look like:

sep-06-2018 16-32-09

### CODE REVIEW

**Comment by BenEmdon** [comment]
This is ready for some 👀 
@d12 @srinjoym

File: app/jobs/repository_import_event_job.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    case status
+    when "success"
+      invite_status.completed!
+      broadcast_assignment_repo_success(user, invite_status, html_url)
+      GitHubClassroom.statsd.increment("v3_exercise_repo.import.success")
+    when "failure"
+      invite_status.errored_importing_starter_code!
+      broadcast_assignment_repo_failure(user, invite_status, html_url)
+      GitHubClassroom.statsd.increment("v3_exercise_repo.import.failure")
```
Comment (BenEmdon) [inline app/jobs/repository_import_event_job.rb]: In the case of failure here I am not destroying the Assignment repo (It's destroyed on manual retry). Should we destroy it here too?

File: app/controllers/application_controller/feature_flags_dependency.rb [hunk 2]
```diff
@@ -37,4 +37,9 @@ def download_repositories_enabled?
     logged_in? && current_user.feature_enabled?(:download_repositories)
   end
   helper_method :download_repositories_enabled?
+
+  def repository_import_webhook_enabled?
+    logged_in? && current_user.feature_enabled?(:repository_import_webhook)
+  end
```
Comment (BenEmdon) [inline app/controllers/application_controller/feature_flags_dependency.rb]: I've added a feature flag so we can roll this out to some % of actors to try it out.

File: app/jobs/repository_import_event_job.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  def perform(payload_body)
+    github_repo_id = payload_body.dig("repository", "id")
+    status = payload_body.dig("status")
+    html_url = payload_body.dig("repository", "html_url")
+
+    repo = AssignmentRepo.find_by(github_repo_id: github_repo_id)
+    return if repo.blank?
+    # Group assignments coming in a follow up PR
+    # repo ||= GroupAssignmentRepo.find_by(github_repo_id: github_repo_id)
```
Comment (BenEmdon) [inline app/jobs/repository_import_event_job.rb]: If we go with this approach I will close #1548 and make this job work for `GroupAssignmentRepo`s too.

File: app/jobs/repository_import_event_job.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    handle_assignment_repo(repo, status, html_url)
+  end
+
+  private
+
+  # rubocop:disable MethodLength
+  def handle_assignment_repo(assignment_repo, status, html_url)
+    user = assignment_repo.user
```
Comment (BenEmdon) [inline app/jobs/repository_import_event_job.rb]: I suppose we should be checking if the user has the feature flag enabled before going ahead with any actions. 
(Will add the feature flag check)

File: app/jobs/repository_import_event_job.rb (hunk 1 above)
Comment (d12) [reply app/jobs/repository_import_event_job.rb]: It's probably ok :+1:

**Comment by BenEmdon** [comment]
I'll wait until Monday to deploy this ⚓️

**Pull request merged**